static/src/javascripts-legacy/projects/commercial/modules/hosted/gallery.js

```diff
@@ -450,32 +450,29 @@ define([
         start();
 
         if (qwery('.js-hosted-gallery-container').length) {
-            loadCssPromise
-            .then(function () {
-                var gallery,
-                    match,
-                    galleryHash = window.location.hash,
-                    res;
-
-                gallery = new HostedGallery();
-                match = /\?index=(\d+)/.exec(document.location.href);
-                if (match) { // index specified so launch gallery at that index
-                    gallery.loadAtIndex(parseInt(match[1], 10));
-                } else {
-                    res = /^#(?:img-)?(\d+)$/.exec(galleryHash);
-                    if (res) {
-                        gallery.loadAtIndex(parseInt(res[1], 10));
+            return loadCssPromise
```

Comment:
ooooh I now see what bothered you. Well, this change wouldn't fly with other commercial modules, but in this case it's fine, as "hosted" modules have no post- or pre- dependency. This change will slightly affect the performance of the commercial init in hosted pages though. A better idea would be for each module to expose an `init` function and separately a collection of non-blocking promises, so that anyone interested can consume their value eventually.